pkg/agent/openflow/client.go

```diff
@@ -417,11 +417,11 @@ func (c *client) UninstallServiceGroup(groupID binding.GroupIDType) error {
 }
 
 func generateEndpointFlowCacheKey(endpointIP string, endpointPort int, protocol binding.Protocol) string {
-	return fmt.Sprintf("Endpoints_%s_%d_%s", endpointIP, endpointPort, protocol)
+	return fmt.Sprintf("E%s%x%s", endpointIP, endpointPort, protocol)
 }
 
 func generateServicePortFlowCacheKey(svcIP net.IP, svcPort uint16, protocol binding.Protocol) string {
-	return fmt.Sprintf("Service_%s_%d_%s", svcIP, svcPort, protocol)
+	return fmt.Sprintf("S%s%x%s", svcIP, svcPort, protocol)
```

Comment:
There will be collision? for example, svcIP="1.1.1.11", svcPort=1  and svcIP="1.1.1.1", svcPort=17: https://play.golang.org/p/aifIXh5WfEE

Reply:
Good point! I changed to: fmt.Sprintf("S%s%s%x", svcIP, protocol, svcPort)